src/arch/xtensa/include/arch/lib/cache.h

```diff
@@ -89,11 +90,11 @@ extern uint32_t _memmap_cacheattr_reset;
 #define is_cached(address) (!!((uintptr_t)(address) & SRAM_UNCACHED_ALIAS))
 #endif
 
-static inline void dcache_writeback_region(void *addr, size_t size)
+static inline void dcache_writeback_region(void __cache *addr, size_t size)
 {
 #if XCHAL_DCACHE_SIZE > 0
 	if (is_cached(addr))
-		xthal_dcache_region_writeback(addr, size);
+		xthal_dcache_region_writeback((__force void *)addr, size);
```

Comment:
Why do we need to force here if addr is already cache attribute ?

Reply:
because `xthal_dcache_region_writeback()` is an XTOS HAL function and we don't want to modify the HAL, right? Obviously the argument is a cached address. But `xthal_dcache_region_writeback()`'s prototype has simply `void *` as the first argument's type